### ISSUE TITLE
Yara DLL

### ISSUE THREAD

**Issue by Tigzy** [VirusTotal/yara#601]
Hello, 
It would be great if you could replace the code here: https://github.com/VirusTotal/yara/blob/master/libyara/include/yara/utils.h#L62

With:
```
#ifdef YR_STATICLIB
#  define YR_API EXTERNC
#elif defined(__GNUC__)
#  define YR_API EXTERNC __attribute__((visibility("default")))
#elif defined(_MSC_VER)
#  if defined(BUILDING_YR)
#    define YR_API  EXTERNC __declspec(dllexport)
#  else
#    define YR_API EXTERNC __declspec(dllimport)
#  endif
#else
#  define YR_API EXTERNC
#endif
```

Then put BUILDING_YR in all project configurations preprocessor
and YR_STATICLIB in .lib generation configurations.

Then, why not create a Debug_DLL and Release_DLL?

**Comment by Tigzy** [comment]
Up?

**Comment by plusvic** [comment]
No, haven't include a configuration to build the DLL, but should be easy make one starting from the exiting ones for building the static library. If you want to create one and send a PR I'll be happy to merge it.
